Design/Workbench/design-projectsaving.md

```diff
@@ -0,0 +1,203 @@
+﻿
+Project Saving for Mantid Workbench
+==============================
+Motivation
+----------
+In MantidPlot it was possible to save all currently open windows and all workspaces, and then be able to load then back in the exact(-ish) state that they were in previously. To achieve this goal again in Mantid Workbench, the problem will be split into two where window serialization and saving and therefore reopening will be separate in implementation, from saving workspaces.
+
+The reason for the split is so that it can be handled in a more abstract manor. Instead of thinking of the task as one great big issue it can be split into the two smaller problems and then split further from there, later on in design.
+
+This document exists because Project Saving is a non-trivial challenge to tackle when it comes to ideas with saving and loading interfaces created in both Python and C++ as well as saving and loading of workspaces. This issue should be fairly simple to understand, simply users should be able to save their projects and load them back in, to at least the standard that MantidPlot was able to achieve.
+
+
+Functional Requirements
+-----------------------
+- Graceful handling of other versions that have been saved previously
```

Comment:
I hope I read this right: the saved projects are properly versioned and can be restored by any future version of Mantid.

Reply:
Not necessarily versioned per say, because anytime a new encoder/decoder would be added or changed it would technically change the version, but the idea is that it should gracefully handle loading an older file.